hack/jenkins/e2e.sh

```diff
@@ -434,6 +434,18 @@ case ${JOB_NAME} in
     : ${PROJECT:="kubernetes-jenkins"}
     ;;
 
+  # Runs slow tests on GCE 1.1 cluster, sequentially.
+  kubernetes-gce-slow-release-1.1)
```

Comment:
having "release" in the name was probably a bad choice, since it isn't really a released version, but rather ci. but I guess we can keep it to be consistent with the other 1.1 e2e job.